### PR TITLE
fix: Converting configuration API response from an array to dictionary

### LINKED ISSUE #1770: Converting configuration API response from an array to dictionary.
## Expected Behavior

This issue is opened to track the corresponding change on components-contrib side for issue dapr/dapr#4556

## Release Note

RELEASE NOTE:

### PR DESCRIPTION
**Pull request by beiwei30**
Signed-off-by: Ian Luo <ian.luo@gmail.com>

# Description

change items from array to dictionary, partially addresses #1770 (other configuration stores need to be updated)

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by beiwei30** [comment]
@LaurenceLiZhixin

File: configuration/redis/redis_test.go [hunk 1]
```diff
@@ -68,14 +68,12 @@ func TestConfigurationStore_Get(t *testing.T) {
...
-					{
-						Key:      "testKey",
+				Items: map[string]*configuration.Item{
+					"testKey": {
 						Value:    "testValue",
 						Metadata: make(map[string]string),
 					},
-				},
-			},
+				}},
```
Comment (daixiang0) [inline configuration/redis/redis_test.go:76]: Please fix lint issue here.

Comment (beiwei30) [reply configuration/redis/redis_test.go]: done.

**Comment by LaurenceLiZhixin** [comment]
LGTM

**Comment by artursouza** [comment]
Let's change this for 1.9 so we have time to update all SDKs. 1.8 is right around the corner.

**Comment by pravinpushkar** [comment]
ping @beiwei30 @berndverst

**Comment by yaron2** [comment]
ping @beiwei30

**Review by berndverst — APPROVED** [review]

**Pull request merged**